server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -237,14 +244,15 @@ private void logToJobConsole(JobIdentifier identifier, String message) {
     }
 
     public boolean shouldAssignWork(ElasticAgentMetadata metadata, String environment, ElasticProfile elasticProfile, ClusterProfile clusterProfile, JobIdentifier identifier) {
-        Map<String, String> clusterProfileProperties = clusterProfile != null ? clusterProfile.getConfigurationAsMap(true) : Collections.EMPTY_MAP;
-        GoPluginDescriptor pluginDescriptor = pluginManager.getPluginDescriptorFor(metadata.elasticPluginId());
-        Map<String, String> configuration = elasticProfile.getConfigurationAsMap(true);
-
-        if (!StringUtils.equals(clusterProfile.getPluginId(), metadata.elasticPluginId())) {
+        if (clusterProfile == null || !StringUtils.equals(clusterProfile.getPluginId(), metadata.elasticPluginId())) {
```

Comment:
In what cases would the `clusterProfile` be `null`, I am guessing for the Azure plugin. Can you please check?
Returning `false` in these conditions might break things?

Reply:
Yes, in case the GoCD server was upgraded and their were few scheduled jobs which now won't have a cluster profile.
No it would not. The check was preset earlier as well. Also, we do a job reload based on timer, which checks for the above scenario and cancels the job with a message. Hence, sending false won't break things.